langchain/schema.py

```diff
@@ -41,10 +41,21 @@ class AgentAction(NamedTuple):
     """Agent's action to take."""
 
     tool: str
-    tool_input: Union[str, dict]
+    tool_input: str
     log: str
 
 
+class StructuredAgentAction(NamedTuple):
```

Comment:
Nit: Any reason to keep on using named tuples instead of data-classes? NamedTuples feel like a python 2 thing

Reply:
I'm not sure honestly, but I'm also not sure what data classes would give us apart from a nice to_dict() method?